File: server/backend/src/main/java/com/canmakan/backend/family/model/FamilyMeResponse.java
package com.canmakan.backend.family.model;

/**
 * Family context for the authenticated caller ({@code GET /api/families/me} and create response).
 * 
 * @author Amelia
 */
public record FamilyMeResponse(
        Long familyId,
        String familyName,
        String memberRole,
        Long selfProfileId,
        Long createdByUserId
) {
}

File: client/mobile/app/src/main/java/sg/edu/nus/iss/canmakan/features/family/data/FamilyMeResponse.kt
package sg.edu.nus.iss.canmakan.features.family.data

import com.google.gson.annotations.SerializedName

/** Domain + wire model for GET /api/families/me (UC8). */
data class FamilyMeResponse(
    @SerializedName("familyId") val familyId: Long,
    @SerializedName("familyName") val familyName: String,
    @SerializedName("memberRole") val memberRole: String,
    @SerializedName("selfProfileId") val selfProfileId: Long,
    @SerializedName("createdByUserId") val createdByUserId: Long,
)
